kratos/spatial_containers/tree.h

```diff
@@ -396,16 +474,35 @@ class Tree
     ///@name Access
     ///@{
 
+    /**
+     * @brief Get a reference to the low point of the bounding box.
+     * @return A reference to the low point of the bounding box.
+     */
     PointType& BoundingBoxLowPoint()
     {
         return mBoundingBoxLowPoint;
     }
 
+    /**
+     * @brief Get a reference to the high point of the bounding box.
+     * @return A reference to the high point of the bounding box.
+     */
     PointType& BoundingBoxHighPoint()
     {
         return mBoundingBoxHighPoint;
     }
 
+    /**
+     * @brief Get the bounding box.
+     * @details This function creates a bounding box using the low and high points and returns it.
+     * @return The bounding box.
+     */
+    BoundingBox<PointType> GetBoundingBox()
```

Comment:
Better to call it `CalculateBoundingBox`, as its not getting the result from the class, its being calculated on every call. (or store it and then return `mBoundingBox`)

Reply:
Better the second option, wait